### ISSUE TITLE
Loading a single-file app's assembly for reflection

### ISSUE THREAD

**Issue by VesCodes** [dotnet/runtime#33270]
Understandably, a self-contained application's published executable is a native host intermediate and not an IL assembly. That said, is there any available or proposed API, or alternative solutions for loading the assembly of a bundled single-file application?

For exercise's sake, my goal is to load an assembly (ideally only for reflection, but that is also not supported in dotnet core) and fetch a custom assembly attribute out of it. The issue is that the target application is published as a single-file executable and as such attempting to `Assembly.Load()` it results in a `bad IL format` error.

**Comment by vitek-karas** [comment]
@asocialraccoon If possible can you please share a bit more specific description of the scenario you're trying to achieve?

I'm asking because lot of our efforts around single-exe are moving the tech in the opposite direction. Crossgen is working on a large-version bubble which will enable AOT precompiled native code for the entire app - fits really nicely with single-exe. From there it's not that far away to move into solutions which don't event persist the original IL fully (Something which for example  .NET Native already does).

It's also anticipated that single-exe will be frequently used together with trimming. The trim process will remove or change parts the IL as well.

Basically this is the first time I'm hearing about a need to run for example ILSpy on a single-exe.

**Comment by swaroop-sridhar** [comment]
CC: @jkotas  
Reg: Considerations for managed APIs to inspect the contents of a single-file bundle.

**Comment by davidfowl** [comment]
> Considerations for managed APIs to inspect the contents of a single-file bundle.

We had some discussions where that would be useful for ASP.NET Core in some scenarios as well.

**Comment by swaroop-sridhar** [comment]
Can you please describe the scenarios? or link the discussions here? Thanks.
